dynamic_library/CMakeLists.txt

```diff
@@ -3,9 +3,12 @@ leatherman_dependency(util)
 if(WIN32)
     leatherman_dependency(windows)
 else()
-  if(NOT CMAKE_SYSTEM_NAME MATCHES "OpenBSD")
+  if(NOT CMAKE_SYSTEM_NAME MATCHES "FreeBSD|OpenBSD")
       add_leatherman_deps(dl)
   endif()
+  if(CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
```

Comment:
This logic should probably be in `cmake/cflags.cmake`? My understanding is that FreeBSD uses GCC or Clang depending on the OS version, and we have support for those compiler explicitly. If something about FreeBSD is causing those flags (including -fPIC) to not get picked up it sould be addressed there.